Automattic/jetpack
# Social: Prevent the metadata updating on autosave

A problem was reported where connections were being incorrectly skipped. It appears that this happens when there is a delay in sync and the post is autosaved in the meantime. If that happens we currently update the connection metadata based on the connections being disabled (which happens immediately after publish).

## Proposed changes:

This is a quick fix for the problem which prevents us from updating the metadata if we're performing an autosave. This is the same behaviour as in the classic editor.





### Other information:


## Jetpack product discussion

p1677557931217669-slack-CDD9LQRSN

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
I'm not sure how to add automated tests for this currently as it requires a bit of setup. but to test it manually:

* Create a post and prepare to publish it with at least one social connection enabled
* Ensure you are sandboxed and have the `async_publish_post` and `async_publicize_post` jobs sandboxed, but do not run the watcher.
* Tail your error log on the sandbox
* Publish the test post, and wait the few seconds it takes for an autosave to happen
* Run the jobs listed in the log.
* After running the `async_publicize_post` job, without this change the connection(s) will be shown as skipped. With this change, it should share the post correctly.
